packages/@romejs/compiler/transforms/compile/transpile/enums.ts

```diff
@@ -0,0 +1,164 @@
+import {
+	AnyJSExpression,
+	AnyJSStatement,
+	JSBindingIdentifier,
+	JSCallExpression,
+	JSFunctionExpression,
+	TSEnumDeclaration,
+	jsNumericLiteral,
+	jsStringLiteral,
+} from "@romejs/ast";
+import {LetBinding, Path, VarBinding} from "@romejs/compiler";
+import {REDUCE_REMOVE} from "@romejs/compiler/constants";
+import {descriptions} from "@romejs/diagnostics";
+import {template, tryStaticEvaluation} from "@romejs/js-ast-utils";
+
+interface PreviousEnumMembers {
+	[name: string]: number | string;
+}
+
+function buildEnumWrapper(
+	id: JSBindingIdentifier,
+	assignments: Array<AnyJSExpression>,
+): AnyJSExpression {
+	const expressionStatement = (template.expression`
+		(function (${id}) {})(${id} || (${id} = {}));
+	` as JSCallExpression);
```

Comment:
For some added safety:

```suggestion
	const expressionStatement = jsCallExpression.assert(template.expression`
		(function (${id}) {})(${id} || (${id} = {}));
	`);
```